### PR TITLE
[Windows/coverage] Fix and Support automaton

### PR DESCRIPTION
**Pull request by gpotter2**
Hey !

I've finally found a way to support Automaton on windows ! That fixes https://github.com/secdev/scapy/issues/451

Every thing except `external_fd` will work exactly as on linux 😄 
`external_fd` will require a `ObjectPipe` instance instead of a `os.pipe` because of the unavailability of `select.select`

Hope it looks good.

@guedou

### CODE REVIEW

**Comment by gpotter2** [comment]
@p-l- Please restart appveyor test

**Comment by gpotter2** [comment]
@p-l- Please restart the appveyor build... It weirdly became stuck

**Comment by p-l-** [comment]
I have restarted the test, but it seems to hang and I have no idea why...

**Comment by gpotter2** [comment]
It looks like it's a delay bug... I'll find a fix

**Comment by gpotter2** [comment]
@p-l- It is a problem of compatibility with coverage :/
https://ci.appveyor.com/project/gpotter2/scapy/build/1.0.201#L185
works

**Comment by p-l-** [comment]
@gpotter2 I don't get it. The build you're linking to also has coverage, hasn't it?

**Comment by gpotter2** [comment]
@p-l- For my tests I replaced 
`%PYTHON%\python -m coverage run -a bin\UTscapy -f text -t test\regression.uts -K mock_read_routes6_bsd || exit /b 42`
with
`%PYTHON%\python bin\UTscapy -f text -t test\regression.uts -K mock_read_routes6_bsd || exit /b 42`

and it worked. Then i've found that `thread` was not supported by coverage, and fixed it

**Comment by p-l-** [comment]
Thanks for you work @gpotter2!

This PR includes a lot of changes, some of them useful, some others not (or at least that I don't understand). Please remove unneeded changes like variable / args rename (`kargs` => `kwargs`, `rd` => `reader`, etc.), lines inverted (example in `ObjectPipe.__init__()`, etc. If I'm wrong about calling these changes unneeded, of course, please explain me.

Please keep in mind that we spend a lot of time reviewing PRs, and each unneeded change means more work for us!

**Comment by gpotter2** [comment]
Sorry for the unneeded changes, and thanks again for your huge work

**Comment by p-l-** [comment]
Thanks again @gpotter2. You may want to leave the `automaton` test category, that can still be useful to developers!

**Comment by gpotter2** [comment]
@p-l- Done :)

File: appveyor.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ test_script:
   - set PATH="%APPVEYOR_BUILD_FOLDER%;C:\Program Files\Wireshark\;%PATH%"
   
   # Main unit tests
-  - "%PYTHON%\\python -m coverage run -a bin\\UTscapy -f text -t test\\regression.uts -F -K automaton -K mock_read_routes6_bsd || exit /b 42"
+  - "%PYTHON%\\python -m coverage run -a bin\\UTscapy -f text -t test\\regression.uts -K mock_read_routes6_bsd || exit /b 42"
```
Comment (p-l-) [inline appveyor.yml]: Please leave the `-F` flag here.

File: scapy/automaton.py [hunk 2]
```diff
@@ -340,20 +364,33 @@ def my_send(self, pkt):
...
+            if WINDOWS:
+                return self.rd.recv(n)
             return os.read(self.rd, n)
         def write(self, msg):
+            if WINDOWS:
+                return self.rd.send(msg)
             return os.write(self.wr,msg)
         def recv(self, n=65535):
-            return self.read(n)        
+            return self.read(n)
```
Comment (p-l-) [inline scapy/automaton.py]: You're not changing this line, please leave it as it is now.

File: scapy/automaton.py [hunk 3]
```diff
@@ -406,7 +443,7 @@ class CommandMessage(AutomatonException):
     ## Services
     def debug(self, lvl, msg):
         if self.debug_level >= lvl:
-            log_interactive.debug(msg)            
+            log_interactive.debug(msg)
```
Comment (p-l-) [inline scapy/automaton.py]: You're not changing this line, please leave it as it is now.

File: scapy/automaton.py [hunk 4]
```diff
@@ -479,7 +518,6 @@ def __init__(self, *args, **kargs):
                     _instance_state(getattr(self, stname)))
         
         self.parse_args(*args, **kargs)
-
```
Comment (p-l-) [inline scapy/automaton.py]: You're not changing this line, please leave it as it is now.

**Comment by gpotter2** [comment]
@p-l- Sorry.

**Pull request merged**